Plex: Always return True when getting auth token if username/password…

disrani opened the pull request:
… aren't configured

This was causing failures to update Plex server when no username and password
were specified since we were failing to get the auth token

SickRage2 commented:
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

fernandog commented on sickbeard/notifiers/plex.py:
```diff
@@ -130,6 +130,7 @@ def update_library(self, ep_obj=None, host=None,  # pylint: disable=too-many-arg
             return False
 
         if not self.get_token(username, password, plex_server_token):
+            logger.log(u'PLEX: Error getting auth token for Plex Media Server, check your settings', logger.ERROR)
```
Warning. Errors can be reported

miigotu replied:
Yeah this should be warning ir not log at all

fernandog commented on sickbeard/notifiers/plex.py:
```diff
@@ -220,7 +221,7 @@ def get_token(self, username=None, password=None, plex_server_token=None):
             return True
 
         if not (username and password):
-            return sickbeard.PLEX_SERVER_NO_AUTH
```
Did you enabled this new setting?

miigotu commented:
I was planning to do this too, but I wanted some verification with the hidden setting I had added first.

disrani commented:
I have changed that to a WARNING, but what is the rule of thumb here for what constitutes an error? I would think not being able to perform a notification should be raised as an error. Without this message a user would have no idea that the notification was not sent.

Also removed the NO_AUTH setting.

fernandog commented:
User action: warning.
Error: issue, error or else that devs need to change code
